lib/gui/os/dialog/services/dialog.js

```diff
@@ -113,20 +113,28 @@ module.exports = function($q, SupportedFormatsModel) {
    * });
    */
   this.showWarning = (options) => {
+    const BUTTONS = [
+      options.confirmationLabel,
+      options.rejectionLabel
+    ];
+
+    const BUTTON_CONFIRMATION_INDEX = _.indexOf(BUTTONS, options.confirmationLabel);
+    const BUTTON_REJECTION_INDEX = _.indexOf(BUTTONS, options.rejectionLabel);
+
     return $q((resolve) => {
       electron.remote.dialog.showMessageBox(currentWindow, {
         type: 'warning',
         buttons: [
           options.confirmationLabel,
           options.rejectionLabel
         ],
-        defaultId: 1,
-        cancelId: 1,
+        defaultId: BUTTON_REJECTION_INDEX,
+        cancelId: BUTTON_REJECTION_INDEX,
         title: 'Attention',
         message: options.title,
         detail: options.description
       }, (response) => {
-        return resolve(response === 0);
+        return resolve(response === BUTTON_CONFIRMATION_INDEX);
```

Comment:
Hm, but that might fail if both the confirmation and rejection label are the same, which would be terrible on its own, but maybe its better to be on the safe side?